contracts/Cashier.sol

```diff
@@ -826,28 +841,158 @@ contract Cashier is ICashier, UsingHelpers, ReentrancyGuard, Ownable, Pausable {
+
+            //Reduce _from escrow amount and increase _to escrow amount
+            escrow[_from] = escrow[_from].sub(totalAmount);
+            escrow[_to] = escrow[_to].add(totalAmount);
+        }
+
+
+        if (paymentType == ETHTKN) {
+
+            //Reduce _from escrow amount and increase _to escrow amount - price
+            escrow[_from] = escrow[_from].sub(price);
+            escrow[_to] = escrow[_to].add(price);
+
+            tokenAddress = IVoucherKernel(voucherKernel).getVoucherDepositToken(
+                tokenSupplyId
+            );
+
+            //Reduce _from escrow token amount and increase _to escrow token amount - deposit
+            escrowTokens[tokenAddress][_from] =  escrowTokens[tokenAddress][_from].sub(depositBu);
+            escrowTokens[tokenAddress][_to] =  escrowTokens[tokenAddress][_to].add(depositBu);
+
+        }
+
+        if (paymentType == TKNETH) {
+            tokenAddress = IVoucherKernel(voucherKernel).getVoucherPriceToken(
+                tokenSupplyId
+            );
+        
+
+            //Reduce _from escrow token amount and increase _to escrow token amount - price 
+            escrowTokens[tokenAddress][_from] =  escrowTokens[tokenAddress][_from].sub(price);
+            escrowTokens[tokenAddress][_to] =  escrowTokens[tokenAddress][_to].add(price);
+
+            //Reduce _from escrow amount and increase _to escrow amount - deposit
+            escrow[_from] = escrow[_from].sub(depositBu);
+            escrow[_to] = escrow[_to].add(depositBu);
+        }
+
+        if (paymentType == TKNTKN) {
+            tokenAddress = IVoucherKernel(voucherKernel).getVoucherPriceToken(
+                tokenSupplyId
+            );
+
+
+            //Reduce _from escrow token amount and increase _to escrow token amount - price 
+            escrowTokens[tokenAddress][_from] =  escrowTokens[tokenAddress][_from].sub(price);
+            escrowTokens[tokenAddress][_to] =  escrowTokens[tokenAddress][_to].add(price);
+
+            tokenAddress = IVoucherKernel(voucherKernel).getVoucherDepositToken(
+                tokenSupplyId
+            );
+
+            //Reduce _from escrow token amount and increase _to escrow token amount - deposit 
+            escrowTokens[tokenAddress][_from] =  escrowTokens[tokenAddress][_from].sub(depositBu);
+            escrowTokens[tokenAddress][_to] =  escrowTokens[tokenAddress][_to].add(depositBu);
+
+        }
+   
+        IBosonRouter(bosonRouterAddress).incrementCorrelationId(_to);
     }
 
+    /**
+     * @notice After the transfer happens the _tokenSupplyId should be updated in the promise. Escrow funds for the seller's deposits (If in ETH) should be allocated to the new owner as well.
+     * @param _from prev owner of the _tokenSupplyId
+     * @param _to nex owner of the _tokenSupplyId
+     * @param _tokenSupplyId _tokenSupplyId for transfer
+     * @param _value qty which has been transferred
+     */
+    function onERC1155Transfer(
```

Comment:
Here as well

Reply:
The hooks were making a lot of calls to the Cashier. Some calls are being made to VouherKernel as well, but most of the external calls were made to Cashier. By moving the hooks to Cashier, the functions can directly acces escrowTokens and escrow mappings. It removes several external calls per hook function, which makes the hook functions more efficient